sdk/identity/identity/src/credentials/managedIdentityCredential/index.ts

```diff
@@ -253,9 +253,15 @@ export class ManagedIdentityCredential implements TokenCredential {
             if (resultToken) {
               logger.info(`SetAppTokenProvider has saved the token in cache`);
               logger.info(`token = ${resultToken.token}`);
+
+              const expiresInSeconds =
+                (resultToken === null)
+                ? 0
+                : ((resultToken.expiresOnTimestamp - Date.now()) / 1000);
```

Comment:
slight tweak here, since `resultToken` is truthy, it can't be `null` 

```suggestion
              const expiresInSeconds =
                resultToken?.expiresOnTimestamp
                ? Math.floor((resultToken.expiresOnTimestamp - Date.now()) / 1000)
                :  0;
```